### PR TITLE
CI: Run auto-milestone on pull_request_target for forks

### PR DESCRIPTION
**Pull request by joshhunt**
Update the Auto Milestone github action to be triggered by `pull_request_target`, which runs with a GITHUB_TOKEN with write permissions. I think this allows PRs from forks to be milestoned.

 - [Permissions for the `GITHUB_TOKEN`](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)
 - [`pull_request_target` trigger docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target)

> This event runs in the context of the base of the pull request, rather than in the context of the merge commit, as the pull_request event does. This prevents execution of unsafe code from the head of the pull request that could alter your repository or steal any secrets you use in your workflow. This event allows your workflow to do things like label or comment on pull requests from forks. Avoid using this event if you need to build or run code from the pull request.

> For workflows that are triggered by the pull_request_target event, the GITHUB_TOKEN is granted read/write repository permission unless the permissions key is specified and the workflow can access secrets, even when it is triggered from a fork. ***Although the workflow runs in the context of the base of the pull request***, you should make sure that you do not check out, build, or run untrusted code from the pull request with this event. Additionally, any caches share the same scope as the base branch. To help prevent cache poisoning, you should not save the cache if there is a possibility that the cache contents were altered. For more information, see "[Keeping your GitHub Actions and workflows secure: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests)" on the GitHub Security Lab website.

Also, this removes the milestone check from pr_checks.json because it'll no longer trigger from changes from GITHUB_TOKEN.

Note... I've had trouble testing this, so I'm open for suggestions on how to be more sure about this! https://github.com/grafana/github-actions-testrepo/pull/123

As this is merged, we'll need an admin to remove the mandatory milestone status check from the repo settings.

### CODE REVIEW

**Comment by hairyhenderson** [comment]
thanks @joshhunt - this might fix the issues we ran into after https://github.com/grafana/grafana/pull/86452

**Review by kminehart — APPROVED** [review]

**Comment by joshhunt** [comment]
Because I haven't been able to test this completely in the test repo, I'm going to wait until Monday to merge this so I don't accidentally block PRs on a very rare sunny London friday.

**Pull request merged**
